### PR TITLE
Some refactoring of list_package.adb and list_sweep.adb

### PR DESCRIPTION
**Pull request by ids1024**

### CODE REVIEW

File: src/commands/list_sweep.adb [hunk 1]
```diff
@@ -170,52 +156,30 @@ procedure List_Sweep (Pa : in out Parse_Array; Pa_Last : in out Integer) is
...
-                  end if;
+                  Allowed := (Pr.IR.Qual.Verb.Tense_Voice_Mood =
+                     (Pres, Active, Ind)) and
+                    ((De.Part.V.Kind in X .. Intrans) or else
+                     (De.Part.V.Kind = Dep) or else
+                     (De.Part.V.Kind = Semidep) or else
+                     (De.Part.V.Kind = Perfdef));
                elsif De.Part.V.Kind = Impers then
-                  if (Pr.IR.Qual.Verb.Person = 3)  and then
+                  Allowed := (Pr.IR.Qual.Verb.Person = 3)  and then
```
Comment (darkestkhan) [inline src/commands/list_sweep.adb:180]: A typo - should be "Allowed =". Assignments are not permitted inside expressions (compile time error).

Comment (mk270) [inline src/commands/list_sweep.adb:180]: Are you sure it is still inside an expression? The diff is hard to read.

Comment (darkestkhan) [inline src/commands/list_sweep.adb:180]: I'm sure - just look at the 3 lines quoted - assignment follows "and then".

Comment (mk270) [inline src/commands/list_sweep.adb:180]: No, it follows "then"; the "and then" appears to be in the line to be deleted. Am I reading this wrong?

Comment (aayjaychan) [inline src/commands/list_sweep.adb:180]: Those lines are assigning a boolean expression to `Allowed`.

The whole statement is

```
                  Allowed := (Pr.IR.Qual.Verb.Person = 3)  and then
                    (Pr.IR.Qual.Verb.Number = S)  and then
                    (Pr.IR.Qual.Verb.Tense_Voice_Mood = (Pres, Active, Ind));
```

Comment (mk270) [inline src/commands/list_sweep.adb:180]: @aayjaychan those aren't the "and then"s @darkestkhan is talking about, surely?

Comment (darkestkhan) [inline src/commands/list_sweep.adb:180]: I'm sorry. Seems like I confused myself.

And as always I forgot one of most important rules - don't read code right
before going to sleep.

**Pull request merged**